packages/compiler/test/checker/references.test.ts

```diff
@@ -535,4 +535,46 @@ describe("compiler: references", () => {
       },
     ]);
   });
+
+  describe("Meta types", () => {
+    describe("ModelProperty::type that is an expression", () =>
+      itCanReference({
+        code: `
+          model Person {
+            address: {
+              @test("target") city: string
+            }
+          }
+        `,
+        ref: "Person.address::type.city",
+      }));
+
+    describe("ModelProperty::type that is a type reference", () =>
+      itCanReference({
+        code: `
+          model Person {
+            address: Address
+          }
+          model Address {
+            @test("target") city: string
+          }
+        `,
+        ref: "Person.address::type.city",
+      }));
+    describe("Operation::returnType", () =>
+      itCanReference({
+        code: `
+          op testOp(): {@test("target")status: 200};
+        `,
+        ref: "testOp::returnType.status",
+      }));
+
+    describe("Operation::parameters", () =>
+      itCanReference({
+        code: `
+          op testOp(@test("target") select: string): void;
```

Comment:
should we have more than one parameter in the operation, just to verify the crrect parameter is selected?